src/Tinfoil/Signing/Ed25519/Internal.hs

```diff
@@ -0,0 +1,128 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE ForeignFunctionInterface #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE GADTs #-}
+module Tinfoil.Signing.Ed25519.Internal (
+    genKeyPair
+  , maxSigLen
+  , pubKeyLen
+  , secKeyLen
+  , signMessage'
+  , verifyMessage'
+) where
+
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+
+import           Foreign (Ptr, Word8, castPtr)
+import           Foreign.C
+import           Foreign.Marshal.Alloc (allocaBytes)
+import           Foreign.Ptr (nullPtr)
+
+import           P
+
+import           System.IO (IO)
+import           System.IO.Unsafe (unsafePerformIO)
+
+import           Tinfoil.Data.KDF
+import           Tinfoil.Data.Key
+
+-- | Generate a new Ed25519 keypair.
+--
+-- FIXME: init requirement for thread safety?
```

Comment:
Nope, tests were failing due to another version difference. Initialization has to do with initializing the RNG and other such things - also important, so this should be fixed before this is merged even though the tests are passing now.